compile warnings on ubuntu 18.08

I get these warnings on ubuntu

```
gcc -DHAVE_CONFIG_H -I. -I..     -D_GNU_SOURCE  -O2 -D_REENTRANT -MT handshake.o -MD -MP -MF .deps/handshake.Tpo -c -o handshake.o handshake.c
handshake.c: In function ‘slave_handshake’:
handshake.c:453:56: warning: ‘%s’ directive output may be truncated writing up to 127 bytes into a region of size 75 [-Wformat-truncation=]
   "Version %s is incompatible with master data version %s",
                                                        ^~
     UPERF_DATA_VERSION, hsp1.version);
                         ~~~~                            
In file included from /usr/include/stdio.h:862:0,
                 from handshake.c:28:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 54 and 181 bytes into a destination of size 128
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mv -f .deps/handshake.Tpo .deps/handshake.Po
gcc -DHAVE_CONFIG_H -I. -I..     -D_GNU_SOURCE  -O2 -D_REENTRANT -MT parse.o -MD -MP -MF .deps/parse.Tpo -c -o parse.o parse.c
mv -f .deps/parse.Tpo .deps/parse.Po
gcc -DHAVE_CONFIG_H -I. -I..     -D_GNU_SOURCE  -O2 -D_REENTRANT -MT shm.o -MD -MP -MF .deps/shm.Tpo -c -o shm.o shm.c
mv -f .deps/shm.Tpo .deps/shm.Po
gcc -DHAVE_CONFIG_H -I. -I..     -D_GNU_SOURCE  -O2 -D_REENTRANT -MT master.o -MD -MP -MF .deps/master.Tpo -c -o master.o master.c
master.c: In function ‘send_command_to_slaves’:
master.c:211:38: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size between 90 and 100 [-Wformat-truncation=]
        "Could not send command %d to %s:%d ",
                                      ^~
In file included from /usr/include/stdio.h:862:0,
                 from master.c:33:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 32 and 307 bytes into a destination of size 128
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
master.c: In function ‘say_goodbye’:
master.c:90:14: warning: ‘%s’ directive output may be truncated writing up to 511 bytes into a region of size between 254 and 509 [-Wformat-truncation=]
        "[%s] %s", p->host, g.message);
              ^~            ~
In file included from /usr/include/stdio.h:862:0,
                 from master.c:33:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 4 and 770 bytes into a destination of size 512
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
master.c:97:14: warning: ‘%s’ directive output may be truncated writing up to 511 bytes into a region of size between 254 and 509 [-Wformat-truncation=]
        "[%s] %s", p->host, g.message);
              ^~            ~
In file included from /usr/include/stdio.h:862:0,
                 from master.c:33:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 4 and 770 bytes into a destination of size 512
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
master.c:102:14: warning: ‘%s’ directive output may be truncated writing up to 511 bytes into a region of size between 254 and 509 [-Wformat-truncation=]
        "[%s] %s\n", p->host, g.message);
              ^~              ~
In file included from /usr/include/stdio.h:862:0,
                 from master.c:33:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 5 and 771 bytes into a destination of size 512
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mv -f .deps/master.Tpo .deps/master.Po

...

gcc -DHAVE_CONFIG_H -I. -I..     -D_GNU_SOURCE  -O2 -D_REENTRANT -MT numbers.o -MD -MP -MF .deps/numbers.Tpo -c -o numbers.o numbers.c
numbers.c: In function ‘adaptive_print_time’:
numbers.c:223:53: warning: ‘__builtin___snprintf_chk’ output may be truncated before the last format character [-Wformat-truncation=]
   (void) snprintf(format, sizeof(format), "%%%d.%dfs", acutal_width, 2);
                                                     ^
In file included from /usr/include/stdio.h:862:0,
                 from numbers.c:23:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 7 and 17 bytes into a destination of size 16
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
numbers.c:227:54: warning: ‘cs’ directive output may be truncated writing 2 bytes into a region of size between 0 and 10 [-Wformat-truncation=]
   (void) snprintf(format, sizeof(format), "%%%d.%df%%cs", acutal_width, 2);
                                                      ^~
In file included from /usr/include/stdio.h:862:0,
                 from numbers.c:23:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 9 and 19 bytes into a destination of size 16
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

--- a/.gitignore
+++ b/.gitignore
@@ -5,3 +5,5 @@ config.status
 .deps
 src/uperf
 src/*.o
+tests/*.log
+tests/*.trs

--- a/src/master.c
+++ b/src/master.c
@@ -77,7 +77,7 @@ static int
 say_goodbye(goodbye_stat_t *total, protocol_t *p, int timeout)
 {
 	goodbye_t g;
-	char msg[GOODBYE_MESSAGE_LEN];
+	char msg[GOODBYE_MESSAGE_LEN + MAXHOSTNAME + 4];
 
 	if (recv_goodbye(&g, p, timeout) != UPERF_SUCCESS) {
 		uperf_error("\nError saying goodbye with %s\n", p->host);
@@ -86,19 +86,19 @@ say_goodbye(goodbye_stat_t *total, protocol_t *p, int timeout)
 
 	switch (g.msg_type) {
 		case MESSAGE_INFO:
-			(void) snprintf(msg, GOODBYE_MESSAGE_LEN,
+			(void) snprintf(msg, sizeof(msg),
 			    "[%s] %s", p->host, g.message);
 			uperf_info(msg);
 			break;
 		case MESSAGE_NONE:
 			break;
 		case MESSAGE_ERROR:
-			(void) snprintf(msg, GOODBYE_MESSAGE_LEN,
+			(void) snprintf(msg, sizeof(msg),
 			    "[%s] %s", p->host, g.message);
 			uperf_log_msg(UPERF_LOG_ERROR, 0, msg);
 			break;
 		case MESSAGE_WARNING:
-			(void) snprintf(msg, GOODBYE_MESSAGE_LEN,
+			(void) snprintf(msg, sizeof(msg),
 			    "[%s] %s\n", p->host, g.message);
 			uperf_log_msg(UPERF_LOG_WARN, 0, msg);
 			break;
@@ -206,7 +206,7 @@ send_command_to_slaves(uperf_cmd cmd, int value)
 	for (i = 0; i < no_slaves; i++) {
 		ret = uperf_send_command(slaves[i], cmd, value);
 		if (ret <= 0) {
-			char msg[128];
+			char msg[MAXHOSTNAME + 64];
 			(void) snprintf(msg, sizeof (msg),
 			    "Could not send command %d to %s:%d ",
 			    value, slaves[i]->host, slaves[i]->port);

--- a/src/numbers.c
+++ b/src/numbers.c
@@ -209,7 +209,7 @@ adaptive_print_time(double value, int width)
 	int acutal_width;
 	int index = 0;
 	static char prefix[] = "num";
-	char format[16];
+	char format[32];
 
 	if (width > 127) {
 		width = 127;

--- a/src/uperf.h
+++ b/src/uperf.h
@@ -19,9 +19,9 @@
 #define		_UPERF_H
 
 /* Keep the data version as 0.2.5 to avoid the version mismatch problem. */
-#define UPERF_DATA_VERSION	"0.3"
+#define UPERF_DATA_VERSION	"0.3.1"
 #define	UPERF_VERSION 		"1.0.7"
-#define	UPERF_VERSION_LEN 	128
+#define	UPERF_VERSION_LEN 	16
 #define	UPERF_EMAIL_ALIAS	"uperf-discuss@lists.sourceforge.net"
 
 #define	MAXHOSTNAME		256

